elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -211,6 +274,68 @@ private static boolean validateSqlInTableConfig(ElideTableConfig elideTableConfi
         return true;
     }
 
+    /**
+     * Validates join clause does not refer to a Table which is not in the same DBConnection.
+     * If joined table is not part of dynamic configuration, then ignore
+     */
+    private static void validateJoinedTablesDBConnectionName(ElideTableConfig elideTableConfig) {
+
+        for (Table table : elideTableConfig.getTables()) {
+            if (!table.getJoins().isEmpty()) {
+
+                Set<String> joinedTables = table.getJoins()
+                        .stream()
+                        .map(join -> join.getTo().toLowerCase(Locale.ENGLISH))
+                        .collect(Collectors.toSet());
+
+                Set<String> connections = elideTableConfig.getTables()
+                        .stream()
+                        .filter(t -> joinedTables.contains(t.getName().toLowerCase(Locale.ENGLISH)))
+                        .map(t -> t.getDbConnectionName())
+                        .collect(Collectors.toSet());
+
+                if (connections.size() > 1 || (connections.size() == 1
+                                && !table.getDbConnectionName().equals(connections.iterator().next()))) {
```

Comment:
Is it possible for DbConnectionName to be null?  I wonder if we should invert this comparison here.

Reply:
DbConnectionName  can not be null as Table schema doesn't allow and default value is Blank("").